syntax = "proto3";
 
 message ServiceRegisterRequest {
    string service_address = 1;
    string service_name = 4;
 }


 message ServiceRegistrationStatusRequest {
     string service_address = 1;
     string service_id = 2;
     string service_name = 3;
 }

 message ServiceRegistrationStatusResponse {
    string bootstrap_service_address = 1;
    string bootstrap_service_id = 2;
    repeated string registered_addresses = 3;
    repeated string registered_ids = 4;
    repeated string registered_server_addresses = 5;
    int32 registered_count = 6;
    bool ready = 7;
 }


 service BootstrapServer {
    rpc RegisterService (ServiceRegisterRequest) returns (ServiceRegistrationStatusResponse) {};
    rpc GetServiceRegistrationStatus (ServiceRegistrationStatusRequest) returns (ServiceRegistrationStatusResponse) {};
 }